### PR TITLE
Add a task to check the host key of a machine

### PR DESCRIPTION
**Pull request by alexmuller**
So that when SSH gives you a scary warning that the remote host identification has changed, you can check what the Puppet master thinks the host key should be.

### CODE REVIEW

File: vm.py [hunk 1]
```diff
@@ -107,3 +107,12 @@ def poweroff():
   from nagios import schedule_downtime
   execute(schedule_downtime, env['host_string'])
   run("sudo poweroff")
+
+@task
+@hosts('puppetmaster-1.management')
+def host_key(hostname):
+  """Check the SSH host key of a machine"""
+  temp_file = '/tmp/ssh_host_key'
```
Comment (dcarley) [inline vm.py]: There could be race condition if two people run this at the same time.

Is it possible to do it without a temporary file? Or store the output of the first command in a variable and then write it to a local temp file using `tempfile.mkstemp()`?

Comment (alexmuller) [inline vm.py]: It doesn't look like `ssh-keygen` is happy to receive things that aren't files, but a bit of Googling has revealed that 3 left chevrons works.

**Comment by timmow** [comment]
I like the `<<<` and `-f /dev/stdin` - for anyone else confused, from the bash manual

```
Here Strings
       A variant of here documents, the format is:

              <<<word

       The word is expanded and supplied to the command on its standard input.
```

Another pattern I like to use to do something similar is the `<()` operator, or [process substitution](http://vincebuffalo.com/2013/08/08/the-mighty-named-pipe.html#process_substitution), although I've found issues when using this with ssh

In particular

`ssh -F <(vagrant ssh-config development-1) development-1` doesnt work

But

`ssh -F /dev/stdin development-1 <<< $(vagrant ssh-config development-1)`

And this is useful for `$REASONS`. 

**TL;DR** `<<<` makes me happy!

**Comment by ajlanghorn** [comment]
This looks good, and I'd have been happy to merge it had I been able to run it from this branch! I've tried `fab preview vm.host_key machine-x`, `fab preview vm.host_key machine-x.vdc` and `fab preview vm.host_key machine-x.vdc.environment` without dice.

**Comment by bradleywright** [comment]
@ajlanghorn would those commands ever work? The normal way to hit a host is `fab preview -H hostname vm.host_key`

**Comment by ajlanghorn** [comment]
Should have explained myself a bit more thoroughly here; pre-caffeine :(
The host_key module takes an argument, `hostname`. Giving a hostname in the usual way with -H returns an error because it's expecting an arg. Poking.

**Comment by bradleywright** [comment]
[`fab task:argument`](http://docs.fabfile.org/en/latest/usage/fab.html#per-task-arguments) is the way to pass arguments. Using a space as a separator is treated as a separate task.

**Comment by dcarley** [comment]
`-H <hostname>` specifies what machine to run the task on. But you don't need to for this task, because it's specified as a `@hosts()` decorator.

Change looks good! Can you find a way to prevent this error if the key isn't found? I think it might happen frequently due to the way we inconsistently use the TLDs in each environment:
```
(fabric)➜  fabric-scripts git:(check-host-key) ✗ fab preview vm.host_key:backend-1.backend
[puppetmaster-1.management] Executing task 'vm.host_key'
[puppetmaster-1.management] out: 2048 24:22:e7:d2:20:0f:02:23:78:e1:00:04:72:f8:ab:2a backend-1.backend.production,backend-1.backend,10.3.0.2 (RSA)
[puppetmaster-1.management] out:

Done.
Disconnecting from puppetmaster-1.management... done.
Disconnecting from jumpbox.preview.alphagov.co.uk... done.
(fabric)➜  fabric-scripts git:(check-host-key) ✗ fab preview vm.host_key:backend-1.backend.preview
[puppetmaster-1.management] Executing task 'vm.host_key'
[puppetmaster-1.management] out: /dev/stdin is not a public key file.
[puppetmaster-1.management] out:

Fatal error: run() received nonzero return code 1 while executing!

Requested: ssh-keygen -l -f /dev/stdin <<< ''
Executed: /bin/bash -l -c "ssh-keygen -l -f /dev/stdin <<< ''"

Aborting.
Disconnecting from puppetmaster-1.management... done.
Disconnecting from jumpbox.preview.alphagov.co.uk... done.
```

**Comment by ajlanghorn** [comment]
Yes, had another look at this; agree that the change looks good and works fine.

**Comment by alexmuller** [comment]
I've added a usage example to the docstring because it wasn't clear. I've also suppressed that error if the SSH key isn't found. Thanks all!

**Comment by dcarley** [comment]
Thanks @alexmuller :+1:
